templates/filebeat.yml

```diff
@@ -13,12 +13,18 @@ filebeat:
       scan_frequency: 10s
       harvester_buffer_size: {{ harvester_buffer_size }}
       max_bytes: {{ max_bytes }}
-      {% if fields -%}
+      {% if fields or principal_unit or model_name -%}
       fields:
         {% for fieldvalue in fields -%}
         {{ fieldvalue.split(':')[0] }}: {{ fieldvalue.split(':')[-1] }}
         {% endfor %}
       {% endif %}
```

Comment:
While it should work as written, this `endif` should go after the `principal_unit` and `model_name` conditions to make it clear that all three entries (fields, principal, model) are guarded by the `if` from line 16.

Otherwise some yahoo may copy pasta the principal and/or model blocks not realizing they belong with the `fields` condition.

Reply:
@kwmonroe good catch! Thank you.